htlcswitch/interceptable_switch.go

```diff
@@ -617,6 +638,49 @@ func (f *interceptedForward) Resume() error {
 	return f.htlcSwitch.ForwardPackets(nil, f.packet)
 }
 
+// ResumeModified resumes the default behavior with field modifications.
+func (f *interceptedForward) ResumeModified(
+	outgoingAmountMsat fn.Option[lnwire.MilliSatoshi],
+	customRecords fn.Option[record.CustomSet]) error {
+
+	// Modify the wire message contained in the packet.
+	htlc, ok := f.packet.htlc.(*lnwire.UpdateAddHTLC)
+	if ok {
+		outgoingAmountMsat.WhenSome(func(amount lnwire.MilliSatoshi) {
+			htlc.Amount = amount
+		})
+
+		//nolint:lll
+		err := fn.MapOptionZ(customRecords, func(records record.CustomSet) error {
+			if len(records) == 0 {
+				return nil
+			}
+
+			// Encode the custom records to bytes.
+			var buf bytes.Buffer
+			if err := records.Encode(&buf); err != nil {
+				return err
+			}
+			recordsBytes := buf.Bytes()
+
+			// Create a new custom records blob TLV record.
+			//nolint:lll
+			tlvRecord := tlv.NewPrimitiveRecord[lnwire.CustomRecordsBlobTlvType](recordsBytes)
+			htlc.CustomRecordsBlob = tlv.SomeRecordT(tlvRecord)
```

Comment:
Should just set `ExtraData`. But then we'd still perform validation to ensure the TLV stream is canonical. So we'd merge the existing extra data that might exist, with what was passed over the RPC.

Reply:
Updated in latest commits.